### PR TITLE
WIP: Deprecate workflow-related keywords (workflow, sequence, parallel, and inlinescript)

### PR DESCRIPTION
**Pull request by KirkMunro**
# PR Summary

Cleanup of workflow code that has been deprecated. In particular:

- deprecation of `workflow` keyword
- deprecation of `parallel` keyword (used for parallel activities in workflow)
- deprecation of `sequence` keyword (used for sequential activities in workflow)
- deprecation of `inlinescript` activity (used for activities outside of workflow)
- removal of some related methods
- removal of deprecated resource strings
- updates to non-deprecated resource strings
- removal of the use of "workflow" in many comments

The keywords in question have been preserved, but will now generate parser errors because they are deprecated.

## PR Context

See issue #9570.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 1]
```diff
@@ -207,7 +207,7 @@ List<CompletionResult> ExecuteGetCommandCommand(bool useModulePrefix)
         }
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-macos`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-windows`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-linux`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

**Comment by KirkMunro** [comment]
@SteveL-MSFT The two Pester errors that occur on each platform happen because `Fabrikam-ServerScript` in `www.poshtestgallery.com` defines a workflow. Since workflow does not parse with this PR in place, and since PowerShellGet wasn't designed with deprecated features that generate parse errors in mind, there's a chicken-and-egg problem with these tests.

Since I can use `Find-Script` in the same PowerShellGet module in a build with this PR to find the script and pull script metadata, which includes workflows that are in the script, we could update PowerShellGet to check for the presence of workflow as part of `Install-Script` (the information is obviously retrievable, and if workflow is found), and prevent the installation from continuing since it is not supported on 7.0+ unless users invoke `Install-Script` with `-Force`. We could then update the failing tests to use `-Force` to install the script even though there are deprecated keywords. This will allow the Pester tests to pass.

We could also update PowerShellGet to recognize parser errors about deprecated keywords. The parser error that is being generated by the parser in this PR is very specific, and clearly identifies when a keyword is deprecated. The keyword token also has a flag indicating that it is deprecated. In the case where a deprecated keyword is found, PowerShellGet could simply ignore that error since it shouldn't care.

You can see both the descriptive error id and the keyword `Deprecated` flag in the output below:

```none
PS C:\> $tokens = $null
>> $errors = $null
>> [System.Management.Automation.Language.Parser]::ParseInput(
>>     'workflow foo {''bar''}; Get-Date',
>>      [ref]$tokens,
>>      [ref]$errors)

Attributes         : {}
UsingStatements    : {}
ParamBlock         :
BeginBlock         :
ProcessBlock       :
EndBlock           : workflow foo {'bar'}; Get-Date
DynamicParamBlock  :
ScriptRequirements :
Extent             : workflow foo {'bar'}; Get-Date
Parent             :


PS C:\> $errors | fl *

Extent          : workflow
ErrorId         : DeprecatedKeywordNotAllowed
Message         : The 'workflow' keyword has been deprecated and is no longer supported.
IncompleteInput : False


PS C:\> $tokens

Text       : workflow
TokenFlags : Keyword, Deprecated
Kind       : Workflow
HasError   : False
Extent     : workflow

Text       : foo
TokenFlags : CommandName
Kind       : Identifier
HasError   : False
Extent     : foo

Text       : {
TokenFlags : ParseModeInvariant
Kind       : LCurly
HasError   : False
Extent     : {

Value      : bar
Text       : 'bar'
TokenFlags : ParseModeInvariant
Kind       : StringLiteral
HasError   : False
Extent     : 'bar'

Text       : }
TokenFlags : ParseModeInvariant
Kind       : RCurly
HasError   : False
Extent     : }

Text       : ;
TokenFlags : ParseModeInvariant
Kind       : Semi
HasError   : False
Extent     : ;

Value      : Get-Date
Text       : Get-Date
TokenFlags : CommandName
Kind       : Generic
HasError   : False
Extent     : Get-Date

Text       :
TokenFlags : ParseModeInvariant
Kind       : EndOfInput
HasError   : False
Extent     :


PS C:\>
```

Please let me know how you would like to proceed.

**Comment by PaulHigin** [comment]
/cc @SteveL-MSFT 
I am not sure if it is worth the effort.  Perhaps we should continue with other workflow cleanup.

**Comment by KirkMunro** [comment]
If we do want these keywords deprecated, the simple solution for PowerShellGet is to check if a script/module has workflow in it _before_ installing it, and prevent that installation from happening if that is the case. That's logical. Users can still save the script, which does not do any parsing checks, and that is the most they would want to do in a version of PowerShell that doesn't support workflow anyway, not even with -Force (why would I be able to install something I can't use -- the installer should prevent that). Since that change should probably happen regardless, and since that change fixes these issues, we could move forward with little effort.

**Comment by iSazonov** [comment]
Today the cleanup is not critical and we could do not break PowerShellGet in the PR. Although we need an understanding of how to proceed in future

**Comment by SteveL-MSFT** [comment]
For now, perhaps we should have an issue opened in PowerShellGet repo to produce an error if a script is incompatible.  For this PR, we can wait on this since it's desirable, but not critical.

**Comment by KirkMunro** [comment]
Issue [PowerShell/PowerShellGet#528](https://github.com/PowerShell/PowerShellGet/issues/528) has been logged so that the issue can be fixed in PowerShellGet.

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-macos`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-linux`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-windows`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

**Comment by iSazonov** [comment]
I set WIP until we PowerShellGet issue will be resolved.

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-linux`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-windows`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-macos`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.158.0/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.158.0/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

**Review by PaulHigin — APPROVED** [review]
These changes look good to me, as long as all tests continue to pass.  Since these include a lot of parser/token changes I would like @daxian-dbw  to also review them.

**Comment by KirkMunro** [comment]
I'm just bumping this to keep it alive.

**Comment by adityapatwardhan** [comment]
@KirkMunro If this is ready for merging, please remove the `WIP` from the title.

@PoshChan please rebuild all

**Comment by adityapatwardhan** [comment]
@PoshChan please rebuild all

**Comment by PoshChan** [comment]
@adityapatwardhan, could not find build at: , error: Invoke-RestMethod : {"$id":"1","innerException":null,"message":"The requested build 36162 could not be found.","typeName":"Microsoft.TeamFoundation.Build.WebApi.BuildNotFoundException, Microsoft.TeamFoundation.Build2.WebApi","typeKey":"BuildNotFoundException","errorCode":0,"eventId":3000}
At D:\home\site\wwwroot\Modules\AzDevOps\AzDevOps.psm1:7 char:9
+         Invoke-RestMethod -Uri $url -Authentication Basic -Credential ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (Method: GET, Reques\u2026PowerShell/6.2.4
}:HttpRequestMessage) [Invoke-RestMethod], HttpResponseException
+ FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand

**Comment by adityapatwardhan** [comment]
Closing and opening PR to trigger CI rebuild

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:207]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-linux`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:207]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-macos`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:207]: @KirkMunro, your last commit had 2 failures in `PowerShell-CI-windows`
PowerShellGet - Script tests (Admin).Should install a script correctly to the required location with AllUsers scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 234
234:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```
PowerShellGet - Script tests.Should install a script correctly to the required location with default CurrentUser scope
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\PowerShellGet\PowerShellGet.Tests.ps1: line 207
207:         $installedScriptInfo | Should -Not -BeNullOrEmpty
```

**Comment by adityapatwardhan** [comment]
@KirkMunro Please fix CI issues and remove `WIP` from title when ready.

**Comment by KirkMunro** [comment]
@adityapatwardhan Here are the problems with this PR:

1. This PR was dependent on another issue (issue [PowerShell/PowerShellGet#528](https://github.com/PowerShell/PowerShellGet/issues/528)). That issue disappeared. I didn't delete it, so how do we find out what happened to that???
1. Assuming that issue gets recovered (or relogged), this PR can't be finished until that issue is addressed (which obviously won't happen if the issue that was logged was deleted).

That is why this was marked WIP: because it is dependent on an issue that I just discovered is missing. I'd like to know what happened to that issue, because issues should not be deleted.

**Comment by adityapatwardhan** [comment]
@alerickson - Can you have a look why the issue in PowerShellGet disappeared?

**Comment by alerickson** [comment]
@adityapatwardhan @KirkMunro the name of the PSGet repository for versions below 3 is now called PowerShellGetv2, the issue is still there though: [PowerShell/PowerShellGetv2/issues/528](https://github.com/PowerShell/PowerShellGetv2/issues/528).  We won't address non-security related issues in that repo, but I'll transfer the issue over to the new repo so we can check before installing scripts in v3.

**Comment by adityapatwardhan** [comment]
@KirkMunro I guess we will have to wait till the version of PSGet that fixes the issue is included in PowerShell release. 
@alerickson Is the fix for the issue going to be part of PSGet v3? or does it not repro in v3 at all?

**Comment by KirkMunro** [comment]
Closing. See https://github.com/PowerShell/PowerShell/pull/10238#issuecomment-826091414.

**Pull request closed without merging**